src/frontend/http.rs

```diff
@@ -1548,12 +1599,24 @@ SELECT
         // CREATE TABLE queries with arrays, so we don't officially support arrays.
 
         let resp = request()
-            .method("POST")
-            .path("/q")
+            .method(method)
+            .path(path)
             .json(&HashMap::from([("query", query)]))
             .reply(&handler)
             .await;
         assert_eq!(resp.status(), StatusCode::OK);
+        assert_eq!(
+            resp.headers()
+                .get(header::CONTENT_TYPE)
+                .unwrap()
+                .to_str()
+                .unwrap(),
+            "application/json; smallint_val=Int16; integer_val=Int32; bigint_val=Int64; char_val=Utf8; \
```

Comment:
It's not a pain to parse on the client, because you can just `.split()` on `;` to get each parameter, and then for each parameter, split on `=` to get the `(key,value)`. The part that is a pain to parse is the types themselves, but that comes down to what we actually want to do with them.

However, I do agree that a JSON encoding of the schema would be easier to parse. It still doesn't solve the issue of what to call (how to interpret) the actual types like `List(...)` though.

Reply:
>  you can just .split() on ; to get each parameter, and then for each parameter, split on = to get the (key,value)

what if my column name contains a semicolon? 🙃

Reply:
> what if my column name contains a semicolon? 🙃

I'm not sure this would be a valid HTTP parameter name, even if it was escaped. :/

Reply:
I didn't find any function/crate for converting Arrow type back to SQL-looking types, so I'd have to make a custom converter. In principle this would be fine for [simple types](https://github.com/apache/arrow-datafusion/blob/26bf31ebbc1274d0f2680d4b944145e794406188/datafusion/sql/src/planner.rs#L304), but for nested and recursive types it would get ugly. Plus, we'd still lose the nullability information regardless.

That's why I went ahead with using the [arrow schema JSON serialization format](https://arrow.apache.org/docs/format/Integration.html#json-test-data-format) now. Note that this is not an official format, it's just something that is used in integration testing, but is consistent. (Also note that I haven't employed url encoding for now.)

The returned schema is very verbose (check out the test assertions), though in theory analytical queries employ a smaller column cardinality (plus we're not "polluting" the actual result output as the schema info is hidden in the header). This could still pose a problem in terms of header size limits; as [per Cloudflare](https://developers.cloudflare.com/workers/platform/limits/#request-limits): `Request headers observe a total limit of 32 KB, but each header is limited to 16 KB`. Maybe we could make the schema type info in the headers optional, so that for some very large output column cardinality people can disable it.

Reply:
For the record, I've further modified this approach in order to url-encode the schema when sending via `content-type` header parameter, as discussed offline.